File: app/src/main/java/com/example/client/screens/login/activity/LoginEmailActivity.java
package com.example.client.screens.login.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.example.client.R;
import com.example.client.screens.login.present.LoginPresent;
import com.example.client.screens.register.activity.RegisterActivity;

public class LoginEmailActivity extends AppCompatActivity implements View.OnClickListener, ILoginView{
    private TextView next,tv_error,register;
    private EditText email;
    private LoginPresent lPresent;
    private LinearLayout view_error;
    private ProgressBar progressBar;
    public static Activity loginEmailActivity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login_email);
        next = findViewById(R.id.next);
        tv_error = findViewById(R.id.tv_error);
        view_error = findViewById(R.id.view_error);
        register = findViewById(R.id.register);
        email = findViewById(R.id.email);
        progressBar = findViewById(R.id.progress_bar);

        loginEmailActivity = this;

        email.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                if(s.length() == 0){
                    setButton(false,R.drawable.bg_btn_disable);
                }
                else {
                    String regex = "^[\\w-\\.]+@([\\w-]+\\.)+[\\w-]{2,4}$";
                    if(email.getText().toString().matches(regex)){
                        setButton(true,R.drawable.bg_btn);
                    }
                    else {
                        setButton(false,R.drawable.bg_btn_disable);
                    }
                }
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        lPresent = new LoginPresent(this);

        next.setOnClickListener(this);
        register.setOnClickListener(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    public void setButton(boolean enable, int background){
        next.setEnabled(enable);
        next.setBackgroundResource(background);
    }


    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.next:
                InputMethodManager imm = (InputMethodManager)getSystemService(Context.INPUT_METHOD_SERVICE);
                if(getCurrentFocus()!=null){
                    imm.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), 0);
                    email.clearFocus();
                }
                lPresent.checkEmail(email.getText().toString());
                break;
            case R.id.register:
                startActivity(new Intent(LoginEmailActivity.this, RegisterActivity.class));
                break;
        }
    }

    @Override
    public void next() {
        view_error.setVisibility(View.GONE);
        Intent intent = new Intent(LoginEmailActivity.this,LoginPasswordActivity.class);
        intent.putExtra("email",email.getText().toString().trim());
        startActivity(intent);
    }

    @Override
    public void login() {

    }

    @Override
    public void showLoading() {
        progressBar.setVisibility(View.VISIBLE);
        next.setBackgroundResource(R.drawable.bg_btn_disable);
        next.setText("");
        next.setEnabled(false);
    }

    @Override
    public void hideLoading() {
        progressBar.setVisibility(View.GONE);
        next.setBackgroundResource(R.drawable.bg_btn);
        next.setText("Tiếp");
        next.setEnabled(true);
    }

    @Override
    public void showErrorMessage(int errMessage) {
        tv_error.setText(getString(errMessage));
        view_error.setVisibility(View.VISIBLE);
    }

}

File: app/src/main/java/com/example/client/screens/login/activity/LoginEmailActivity.kt
package com.example.client.screens.login.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.view.inputmethod.InputMethodManager
import com.example.client.R
import com.example.client.app.Constants
import com.example.client.base.BaseActivityMVP
import com.example.client.models.event.Event
import com.example.client.screens.login.present.ILoginPresent
import com.example.client.screens.login.present.LoginPresent
import com.example.client.screens.register.activity.RegisterActivity
import kotlinx.android.synthetic.main.activity_login_email.*

class LoginEmailActivity : BaseActivityMVP<ILoginPresent>(), View.OnClickListener, ILoginView {

    companion object {
        @JvmStatic
        fun newInstance(from: Activity): Intent = Intent(from, LoginEmailActivity::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login_email)
    }

    override fun bindEvent() {
        et_email.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                if (s.isEmpty()) {
                    setButton(false, R.drawable.bg_btn_disable)
                } else {
                    val regex = "^[\\w-.]+@([\\w-]+\\.)+[\\w-]{2,4}$".toRegex()
                    if (et_email.text.toString().matches(regex)) {
                        setButton(true, R.drawable.bg_btn)
                    } else {
                        setButton(false, R.drawable.bg_btn_disable)
                    }
                }
            }

            override fun afterTextChanged(s: Editable) {}
        })
        tv_next.setOnClickListener(this)
        tv_register.setOnClickListener(this)
    }

    fun setButton(enable: Boolean, background: Int) {
        tv_next.isEnabled = enable
        tv_next.setBackgroundResource(background)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_next -> {
                val imm = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
                currentFocus?.let {
                    imm.hideSoftInputFromWindow(it.windowToken, 0)
                    et_email.clearFocus()
                }
                presenter.checkEmail(et_email.text.toString())
            }
            R.id.tv_register -> startActivity(Intent(this@LoginEmailActivity, RegisterActivity::class.java))
        }
    }

    override fun next() {
        lnl_error.visibility = View.GONE
        startActivity(LoginPasswordActivity.newInstance(this, email = et_email.text.toString().trim()))
    }

    override fun loginSuccess() {}
    override fun showLoading() {
        rll_loading.visibility = View.VISIBLE
    }

    override fun hideLoading() {
        rll_loading.visibility = View.GONE
    }

    override fun onBackPress() {
        super.onBackPressed()
    }

    override fun showErrorMessage(errMessage: Int) {
        tv_error.text = getString(errMessage)
        lnl_error.visibility = View.VISIBLE
    }

    override val presenter: ILoginPresent
        get() = LoginPresent(this)

}